### PR TITLE
Properly escape appPath under Linux and fix appPath with AppImage

### PR DESCRIPTION
**Pull request by Oxalin**
- Target platforms this affects (Linux, Mac, Mac app store, and or Windows): Linux
- What problem does this solve? path containing spaces are not properly escaped and mountpoint of AppImage is used instead of its actual path.
- Could it break any existing functionality for users? No, the code is limited to Linux where it is already not working properly.

---

The fix escapes properly paths containing spaces works fine. However, according to the Desktop Entry Specification (https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html), this shouldn't be limited to spaces, but to every reserved characters. Also, the entry should be quoted. This needs a bit more test, mostly when combined with arguments.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,9 +45,14 @@
   "homepage": "https://github.com/4ver/node-auto-launch",
   "dependencies": {
     "applescript": "^1.0.0",
+    "gulp-rimraf": "^1.0.0",
     "mkdirp": "^0.5.1",
+    "natives": "^1.1.6",
```
Comment (4ver) [inline package.json:50]: Is this needed?

File: package.json [hunk 2]
```diff
@@ -44,9 +45,14 @@
   "homepage": "https://github.com/4ver/node-auto-launch",
   "dependencies": {
     "applescript": "^1.0.0",
+    "gulp-rimraf": "^1.0.0",
```
Comment (4ver) [inline package.json:48]: Should be in dev dependencies

File: src/index.coffee [hunk 3]
```diff
@@ -23,10 +23,11 @@ module.exports = class AutoLaunch
         versions = process?.versions
         if path?
             # Verify that the path is absolute
-            throw new Error 'path must be absolute' unless isPathAbsolute path
+            throw new Error 'path must be absolute' unless pathTools.isAbsolute path
```
Comment (4ver) [inline src/index.coffee:26]: Looks like `path.isAbsolute` was introduced into node.js in version 11. I think we have support version 8 and up.

Best to leave it as is on this pr unless it's necessary. Be nice to drop support for really old versions of node but but as a specific change and major release.

**Review by 4ver — CHANGES_REQUESTED** [review]

File: src/index.coffee [hunk 4]
```diff
@@ -1,16 +1,16 @@
-isPathAbsolute = require 'path-is-absolute'
```
Comment (4ver) [inline src/index.coffee:1]: Remove the dependency from package.json

**Review by 4ver — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (Oxalin) [reply package.json]: fixed by https://github.com/Oxalin/node-auto-launch/commit/e46e9092cf84d0c06570bbcc621bb96358c67290

File: src/index.coffee (hunk 4 above)
Comment (Oxalin) [reply src/index.coffee:1]: fixed by https://github.com/Oxalin/node-auto-launch/commit/e46e9092cf84d0c06570bbcc621bb96358c67290

File: package.json (hunk 1 above)
Comment (Oxalin) [reply package.json]: Doesn't seem so. Fixed by https://github.com/Oxalin/node-auto-launch/commit/e4e637dc1e94173cc04995a17e9a54664d1e9d4a

**Pull request merged**